homeassistant/components/sensor/deconz.py

```diff
@@ -0,0 +1,165 @@
+"""
+Support for deCONZ sensor support.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/sensor/deconz/
+"""
+
+import asyncio
+import logging
+
+from homeassistant.components.deconz import (
+    ATTR_FW_VERSION, ATTR_MANUFACTURER, ATTR_MODEL_ID, ATTR_UNIQUE_ID,
+    CONF_SWITCH_AS_EVENT, DECONZ_DATA, DOMAIN, DeconzEvent)
+from homeassistant.const import ATTR_BATTERY_LEVEL
+from homeassistant.core import callback
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.icon import icon_for_battery_level
+from homeassistant.util import slugify
+
+DEPENDENCIES = [DOMAIN]
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_EVENT_ID = 'event_id'
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Setup sensor platform for Deconz."""
+    from pydeconz.sensor import DECONZ_SENSOR
+    switch_as_event = discovery_info.get(CONF_SWITCH_AS_EVENT)
+    sensors = hass.data[DECONZ_DATA].sensors
+
+    for _, sensor in sensors.items():
+        if sensor.type in DECONZ_SENSOR:
+            if switch_as_event and sensor.type == 'ZHASwitch':
+                DeconzEvent(hass, sensor)
+                if sensor.battery:
+                    async_add_devices([DeconzBattery(sensor)], True)
```

Comment:
We should not have batteries as standalone sensors. Instead, add a ATTR_BATTERY_LEVEL to the switch.

Reply:
@balloob as a user (i was the one requesting this) i like that battery level are separate sensors. I think this makes it much easier to show a list of battery level instead of creating a template sensor for each one to pull the attribute out and show it in the dashboard, is there an easier way to show this ? I noticed for instance that my iphone battery level is added as a separate sensor in HASS, is that not the same thing ?

Reply:
Balloob, this is what we talked about a few weeks ago on discord. Switch is only for remotes, not a physical wall switch with multiple states.